### PR TITLE
split nightly tests, too

### PR DESCRIPTION
**Pull request by jstriebel**
### Description:
- Split the nightly tests similar to the normal CI runs, to avoid being terminated due to unknown reasons

### CODE REVIEW

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -3,6 +3,7 @@ name: nightly
 on:
   schedule:
     - cron:  '00 06 * * *'
+  pull_request: {}
```
Comment (jstriebel) [inline .github/workflows/nightly.yml:6]: TODO: remove before merging

Comment (jstriebel) [reply .github/workflows/nightly.yml:6]:
```suggestion
```

File: .github/workflows/nightly.yml [hunk 2]
```diff
@@ -40,15 +41,9 @@ jobs:
     - name: Python tests, refreshing the network snapshots
       env:
         WK_TOKEN: ${{ secrets.WK_TOKEN }}
-      run: ./test.sh --refresh-snapshots
+      run: ./test.sh --refresh-snapshots --splits 5 --group ${{ matrix.group }}
```
Comment (normanrz) [inline .github/workflows/nightly.yml:44]: Shouldn't this be 3?
```suggestion
      run: ./test.sh --refresh-snapshots --splits 3 --group ${{ matrix.group }}
```

Comment (jstriebel) [reply .github/workflows/nightly.yml:44]: Yep! This is also a problem in the normal CI action, fixing it there as well :see_no_evil:

File: .github/workflows/nightly.yml [hunk 3]
```diff
@@ -40,15 +41,9 @@ jobs:
...
       env:
         WK_TOKEN: ${{ secrets.WK_TOKEN }}
-      run: ./test.sh --refresh-snapshots
-
-    - name: Check if git is dirty
-      continue-on-error: true  # This is currently expected to fail, might change in the future
-      run: |
-        git diff --no-ext-diff --quiet --exit-code
-        [[ -z $(git status -s) ]]
+      run: ./test.sh --refresh-snapshots --splits 5 --group ${{ matrix.group }}
```
Comment (jstriebel) [inline .github/workflows/nightly.yml:49]:
```suggestion
      run: ./test.sh --refresh-snapshots --splits 3 --group ${{ matrix.group }}
```

**Review by normanrz — APPROVED** [review]

**Pull request merged**